core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -129,6 +132,15 @@ public DispatchQuery createDispatchQuery(
                 return queryExecutionFactory.createQueryExecution(preparedQuery, stateMachine, slug, warningCollector);
             }
             catch (Throwable e) {
+                if (e instanceof StackOverflowError) {
```

Comment:
shouldn't the emergency catch cover all Errors?

Reply:
I was thinking the same. I filter all Errors in other place. But I think this place should do that too.

Reply:
Changed